src/java.base/share/classes/java/lang/Object.java

```diff
@@ -33,6 +33,25 @@
  * Every class has {@code Object} as a superclass. All objects,
  * including arrays, implement the methods of this class.
  *
+ * @apiNote
+ * <h2><a id=objectOverrides>Guidance for overriding {@code Object}
+ * methods</a></h2>
+ *
+ * {@code java.lang.Object} includes three {@code public} methods
+ * overridable in subclasses, {@link #equals(Object)}, {@link
+ * #hashCode()}, and {@link #toString()}. Each of these methods can be
+ * called by any code that has access to an object of a
+ * class. Especially if an object can be used by code in a separate
+ * maintenance domain, any overrides of these methods should take care
+ * to provide reasonable implementations of these methods. In
+ * particular, the implementations should take care to avoid using
+ * excessive memory, computational time, or any other resources.
+ * Additionally, during typical operation these methods should
```

Comment:
A note on terminology, I used "typical" for the new notes rather than "normal" to avoid conflating the discussion with the JLS concept of code "completing normally" (as opposed to completing abruptly due to an exception):

https://docs.oracle.com/javase/specs/jls/se22/html/jls-14.html#jls-14.1